### PR TITLE
AIP-84 Get Mapped Task Instance Try Details

### PR DESCRIPTION
**Pull request by kandharvishnu**
related to: https://github.com/apache/airflow/issues/42370

---

### CODE REVIEW

File: tests/api_fastapi/core_api/routes/public/test_task_instances.py [hunk 1]
```diff
@@ -1497,6 +1497,75 @@ def test_should_respond_200_with_different_try_numbers(self, test_client, try_nu
             "dag_run_id": "TEST_DAG_RUN_ID",
         }
 
+    @pytest.mark.parametrize("try_number", [1, 2])
+    def test_should_respond_200_with_mapped_task_at_different_try_numbers(
+        self, test_client, try_number, session
+    ):
+        tis = self.create_task_instances(session, task_instances=[{"state": State.FAILED}])
+        print(tis)
```
Comment (rawwar) [inline tests/api_fastapi/core_api/routes/public/test_task_instances.py]:
```suggestion
```

File: tests/api_fastapi/core_api/routes/public/test_task_instances.py [hunk 2]
```diff
@@ -1497,6 +1497,75 @@ def test_should_respond_200_with_different_try_numbers(self, test_client, try_nu
...
+        old_ti = tis[0]
+        print(old_ti)
```
Comment (rawwar) [inline tests/api_fastapi/core_api/routes/public/test_task_instances.py]:
```suggestion
```

File: tests/api_fastapi/core_api/routes/public/test_task_instances.py [hunk 3]
```diff
@@ -1497,6 +1497,75 @@ def test_should_respond_200_with_different_try_numbers(self, test_client, try_nu
...
+        for idx in (1, 2):
+            ti = TaskInstance(task=old_ti.task, run_id=old_ti.run_id, map_index=idx)
+            ti.rendered_task_instance_fields = RTIF(ti, render_templates=False)
+            ti.try_number = 1
+            for attr in ["duration", "end_date", "pid", "start_date", "state", "queue", "note"]:
+                setattr(ti, attr, getattr(old_ti, attr))
+            session.add(ti)
+        session.commit()
+        tis = session.query(TaskInstance).all()
+        print(tis)
```
Comment (rawwar) [inline tests/api_fastapi/core_api/routes/public/test_task_instances.py]:
```suggestion
```

File: tests/api_fastapi/core_api/routes/public/test_task_instances.py [hunk 4]
```diff
@@ -1497,6 +1497,75 @@ def test_should_respond_200_with_different_try_numbers(self, test_client, try_nu
...
+        clear_task_instances(tis, session)
+        # Simulate the try_number increasing to new values in TI
+        for ti in tis:
+            if ti.map_index > 0:
+                ti.try_number += 1
+                ti.queue = "default_queue"
+                session.merge(ti)
+        session.commit()
+        tis = session.query(TaskInstance).all()
+        print(tis)
```
Comment (rawwar) [inline tests/api_fastapi/core_api/routes/public/test_task_instances.py]:
```suggestion
```

File: tests/api_fastapi/core_api/routes/public/test_task_instances.py [hunk 5]
```diff
@@ -1497,6 +1497,75 @@ def test_should_respond_200_with_different_try_numbers(self, test_client, try_nu
...
+        # in each loop, we should get the right mapped TI back
+        for map_index in (1, 2):
+            # Get the info from TIHistory: try_number 1, try_number 2 is TI table(latest)
+            print(
+                "/public/dags/example_python_operator/dagRuns/TEST_DAG_RUN_ID/taskInstances"
+                f"/print_the_context/{map_index}/tries/{try_number}",
+            )
```
Comment (rawwar) [inline tests/api_fastapi/core_api/routes/public/test_task_instances.py]:
```suggestion
```

File: tests/api_fastapi/core_api/routes/public/test_task_instances.py [hunk 6]
```diff
@@ -1497,6 +1497,75 @@ def test_should_respond_200_with_different_try_numbers(self, test_client, try_nu
...
+            response = test_client.get(
+                "/public/dags/example_python_operator/dagRuns/TEST_DAG_RUN_ID/taskInstances"
+                f"/print_the_context/{map_index}/tries/{try_number}",
```
Comment (rawwar) [inline tests/api_fastapi/core_api/routes/public/test_task_instances.py:1533]: can you add a comment here that once Authentication is added, we need to update this to pass user details like we are doing in legacy

**Review by rawwar — CHANGES_REQUESTED** [review]
can you remove prints

File: tests/api_fastapi/core_api/routes/public/test_task_instances.py (hunk 5 above)
Comment (rawwar) [inline tests/api_fastapi/core_api/routes/public/test_task_instances.py]:
```suggestion
```

File: tests/api_fastapi/core_api/routes/public/test_task_instances.py (hunk 4 above)
Comment (rawwar) [inline tests/api_fastapi/core_api/routes/public/test_task_instances.py]:
```suggestion
```

**Review by pierrejeambrun — COMMENTED** [review]
Looking good overall beside @rawwar comments.

**Comment by kandharvishnu** [comment]
> Looking good overall beside @rawwar comments.

yes, removed the print statements

**Review by pierrejeambrun — APPROVED** [review]
Nice thanks.

**Review by rawwar — APPROVED** [review]

**Comment by rawwar** [comment]
Just noticed logical_date is missing in response model(TaskInstanceHistoryResponse). Is this intentional ?

**Comment by kandharvishnu** [comment]
> Just noticed logical_date is missing in response model(TaskInstanceHistoryResponse). Is this intentional ?

TaskInstanceHistoryResponse does not have `logical_date` 
image

**Comment by rawwar** [comment]
I saw the response [Here](https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html#operation/get_mapped_task_instance_try_details) and saw it had execution_date

**Pull request merged**